experimental/bridge/opentracing/lib.go

```diff
@@ -0,0 +1,399 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentracing
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"google.golang.org/grpc/codes"
+
+	ot "github.com/opentracing/opentracing-go"
+	otext "github.com/opentracing/opentracing-go/ext"
+	otlog "github.com/opentracing/opentracing-go/log"
+
+	otelcore "go.opentelemetry.io/api/core"
+	otelevent "go.opentelemetry.io/api/event"
+	otelregistry "go.opentelemetry.io/api/registry"
+	oteltrace "go.opentelemetry.io/api/trace"
+)
+
+type bridgeSpanContext struct {
+	// TODO: have a look at the java implementation of the shim to
+	// see what do they do with the baggage items
+	baggageItems    map[string]string
+	otelSpanContext otelcore.SpanContext
+}
+
+var _ ot.SpanContext = &bridgeSpanContext{}
+
+func (c *bridgeSpanContext) ForeachBaggageItem(handler func(k, v string) bool) {
+	for k, v := range c.baggageItems {
+		if !handler(k, v) {
+			break
+		}
+	}
+}
+
+func (c *bridgeSpanContext) setBaggageItem(restrictedKey, value string) {
+	if c.baggageItems == nil {
+		c.baggageItems = make(map[string]string)
+	}
+	c.baggageItems[restrictedKey] = value
+}
+
+func (c *bridgeSpanContext) baggageItem(restrictedKey string) string {
+	return c.baggageItems[restrictedKey]
+}
+
+type bridgeSpan struct {
+	otelSpan oteltrace.Span
+	ctx      *bridgeSpanContext
+	tracer   *bridgeTracer
+}
+
+var _ ot.Span = &bridgeSpan{}
+
+func (s *bridgeSpan) Finish() {
+	s.otelSpan.Finish()
+}
+
+func (s *bridgeSpan) FinishWithOptions(opts ot.FinishOptions) {
+	// FIXME: I can't see how we can support the FinishTime option.
+	//
+	// TODO: Do we handle LogRecords by calling Event for all the
+	// items in the array? Same with BulkLogData?
+	s.Finish()
+}
+
+func (s *bridgeSpan) Context() ot.SpanContext {
+	return s.ctx
+}
+
+func (s *bridgeSpan) SetOperationName(operationName string) ot.Span {
+	// TODO: Is this about changing the name of the span? If so,
+	// not sure it's possible with otel.
+	return s
+}
+
+func (s *bridgeSpan) SetTag(key string, value interface{}) ot.Span {
+	switch key {
+	case string(otext.SpanKind):
+		// TODO: Should we ignore it?
+	case string(otext.Error):
+		if b, ok := value.(bool); ok {
+			status := codes.OK
+			if b {
+				status = codes.Unknown
+			}
+			s.otelSpan.SetStatus(status)
+		}
+	default:
+		s.otelSpan.SetAttribute(otTagToOtelCoreKeyValue(key, value))
+	}
+	return s
+}
+
+func (s *bridgeSpan) LogFields(fields ...otlog.Field) {
+	// TODO: The context shouldn't be stored anywhere, so
```

Comment:
There's a philosophical debate about whether you should ever store a `context.Context` in a struct. https://golang.org/pkg/context/ says Do Not do this, but I think it's OK. I would store the `context.Context` from the span `Start` event in order to use here.